File: fis-nutricion-infantil/backend/src/main/java/backend/service/InfanteService.java
package backend.service;

import backend.model.Infante;
import backend.repository.InfanteRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class InfanteService {
    private final InfanteRepository infanteRepository;

    public InfanteService(InfanteRepository infanteRepository) {
        this.infanteRepository = infanteRepository;
    }

    public Infante guardar(Infante infante) {
        return infanteRepository.save(infante);
    }

    public List<Infante> listar() {
        return infanteRepository.findAll();
    }

    public Optional<Infante> buscarPorId(Integer id) {
        return infanteRepository.findById(id);
    }

    public void eliminar(Integer id) {
        infanteRepository.deleteById(id);
    }
}

File: fis-nutricion-infantil/frontend/src/services/infanteService.js
import apiClient from './apiClient'

export const infanteService = {
  listar: () =>
    apiClient.get('/infantes').then((response) => response.data),

  buscarPorId: (id) =>
    apiClient.get(`/infantes/${id}`).then((response) => response.data),

  guardar: (data) =>
    apiClient.post('/infantes', data).then((response) => response.data),

  actualizar: (id, data) =>
    apiClient.put(`/infantes/${id}`, data).then((response) => response.data),

  eliminar: (id) =>
    apiClient.delete(`/infantes/${id}`).then((response) => response.data),
}